### PR TITLE
rustup-init: Depends on openssl

### LINKED ISSUE #3143: rustup-init: error: failed to run custom build command for `openssl-sys v0.9.6`
# Please always follow these steps:

- [x] Confirmed this is a problem with `brew install`ing one, specific formula and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Linuxbrew/brew/issues/new.
- [ ] Attempt to replicate the problem on macOS with Homebrew if possible. If you can replicate the problem on macOS with Homebrew, please report the issue to [Homebrew/core](https://github.com/Homebrew/homebrew-core). If you are unsure, please report the issue to Linuxbrew.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

**Please note we will close your issue without comment if you delete or do not fill out the issue checklist and provide ALL the requested information.**

To help us debug your issue please explain:
- What you were trying to do (and why)
  brew install rustup
- What happened (include command output)
 build failed, see gist with output
- What you expected to happen
  it compiles & works
- Step-by-step reproduction instructions (by running `brew install` commands)
  brew install rust && brew install rustup

See gist for full output
https://gist.github.com/dant3/8b0d4d15bb9093e53116d78ac58766d9

### PR DESCRIPTION
**Pull request by dant3**
- [x] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/rustup-init.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ class RustupInit < Formula
   end
 
   depends_on "rust" => :build
+  depends_on "openssl" => :build
```
Comment (sjackman) [inline Formula/rustup-init.rb]: Please add `unless OS.mac?` to the end of this line.

**Comment by sjackman** [comment]
The preferred first line of the git commit log message is…
```
rustup-init: Depends on openssl for Linuxbrew
```
You can put whatever you like in the body.

**Comment by dant3** [comment]
@sjackman I've fixed everything you've mentioned. However build on CI is not passing and I'm not sure why.
```
==> brew install --only-dependencies --build-bottle --verbose rustup-init
rust: CIRequirement unsatisfied!
Error: An unsatisfied requirement failed this build.
==> FAILED
==> brew install --build-bottle --verbose rustup-init
rust: CIRequirement unsatisfied!
Error: An unsatisfied requirement failed this build.
==> FAILED
```
What kind of unsatisfied requirement happens here? If I do same thing manually on my machine it works just fine.

**Comment by rwhogg** [comment]
@dant3 We specifically disable certain formulae from building on our CI (Travis and Circle) systems; that's what the `CIRequirement` is.

The requirement that's unsatisfied is coming from `rust`, not `rustup-init`.

We do this when formulae take too long to compile and/or exhaust all of Circle CI's available memory.

I'm currently building `rust` locally and will try building `rustup-init` off of your branch when that's done; if it's successful I'll merge this PR later tonight.

I apologize for the frustrations our CI system causes contributors. We're trying to make it better.

**Comment by rwhogg** [comment]
Thanks a bunch for your contribution to Linuxbrew @dant3! We appreciate it 😸

**Comment by sjackman** [comment]
@rwhogg Did you build `rust` locally with `brew test-bot-docker`, and would you like to push that bottle?

**Comment by rwhogg** [comment]
No, I didn't build a bottle for `rust`. I can try tonight, no guarantees it will work though.

**Pull request merged**